src/AWS.Deploy.Common/Recipes/Validation/RecipeValidators/BeanstalkInstanceTypeValidator.cs

```diff
@@ -0,0 +1,132 @@
+                    return await ValidationResult.FailedAsync("Could not find a valid value for Environment Name " +
+                    "as part of of the Elastic Beanstalk deployment configuration. Please provide a valid value and try again.");
+            }
+            catch (OptionSettingItemDoesNotExistException)
+            {
+                return await ValidationResult.FailedAsync("Could not find a valid value for Environment Name " +
+                    "as part of of the Elastic Beanstalk deployment configuration. Please provide a valid value and try again.");
+            }
+
+            // In order to retrieve the Instance Type from Elastic Beanstalk, we need to list the Option Settings of the configuration settings and look for the option
+            // with the namespace 'aws:autoscaling:launchconfiguration' and name 'InstanceType'.
+            var environmentSettings = await _awsResourceQueryer.DescribeElasticBeanstalkConfigurationSettings(applicationName, environmentName);
+            var environmentInstanceTypes = new HashSet<string>();
+            foreach (var environmentSetting in environmentSettings)
+            {
+                foreach (var optionSetting in environmentSetting.OptionSettings)
+                {
+                    if (optionSetting.Namespace.Equals("aws:autoscaling:launchconfiguration") &&
+                        optionSetting.OptionName.Equals("InstanceType"))
+                    {
+                        environmentInstanceTypes.Add(optionSetting.Value);
+                    }
+                }
+            }
+
+            // Once we have the instance types, we need to retrieve the supported architecture for those instance types.
+            var environmentArchitectures = new HashSet<string>();
+            foreach (var environmentInstanceType in environmentInstanceTypes)
+            {
+                var describeInstanceTypeResponse = await _awsResourceQueryer.DescribeInstanceType(environmentInstanceType);
+                describeInstanceTypeResponse?.ProcessorInfo.SupportedArchitectures.ForEach((x) => environmentArchitectures.Add(x));
+            }
+
+            // We check if the selected instance types support the architecture we are trying to deploy to.
+            if (!environmentArchitectures.Contains(recommendation.DeploymentBundle.EnvironmentArchitecture.ToString(), StringComparer.InvariantCultureIgnoreCase))
+            {
+                return await ValidationResult.FailedAsync(
+                    $"The Elastic Beanstalk application is currently using the Instance Types '{string.Join(",", environmentInstanceTypes)}' " +
+                    $"which do not support the currently selected Environment Architecture '{recommendation.DeploymentBundle.EnvironmentArchitecture}'. " +
+                    "Please select an Instance Type that supports the currently selected Environment Architecture.");
+            }
+        }
+        else

```

Comment:
this `else` statement handles 3 scenarios right?

1. the user tries to change the instance type without changing the architecture
2. The user tries to change the architecture without changing the instance type (but they still explicitly set the existing one)
3. the user tries to change the architecture and instance type at the same time

Reply:
Yes that is true, but more broadly this always checks that the instance type and environment architecture are compatible in the case where an instance type is explicitly defined.